services/app-api/src/resolvers/healthPlanPackage/updateHealthPlanFormData.ts

```diff
@@ -170,49 +170,28 @@ export function updateHealthPlanFormDataResolver(
                 })
             }
 
-            // Check if draft revision exist, if not then return error
+            // If contract is draft and draft revision does not exist, then possibly:
+            // - contract was mislabeled labelled DRAFT in the parsing functions.
+            // - something happened in the converter functions to not add draft revision.
+            // - there were no revisions without submitted info.
             if (!contractWithHistory.draftRevision) {
                 const errMessage = `Issue finding a draft revision for contract id ${input.pkgID}. Message: Draft revision not found}`
                 throw new GraphQLError(errMessage, {
                     extensions: {
-                        code: 'NOT_FOUND',
-                        cause: 'DB_ERROR',
+                        code: 'INTERNAL_SERVER_ERROR',
                     },
                 })
             }
 
-            //check for fields
-            // Validate that none of these protected fields have been modified.
-            // These fields should only be modified by the server, never by an update.
-            const previousFormData = {
-                id: contractWithHistory.draftRevision.id,
-                stateCode: contractWithHistory.stateCode,
-                stateNumber: contractWithHistory.stateNumber,
-                // Matches the format of the created at date that comes out of the proto.
-                createdAt: new Date(
-                    Date.UTC(
-                        contractWithHistory.draftRevision.updatedAt.getUTCFullYear(),
-                        contractWithHistory.draftRevision.updatedAt.getUTCMonth(),
-                        contractWithHistory.draftRevision.updatedAt.getUTCDate()
-                    )
-                ),
-                updatedAt: contractWithHistory.draftRevision.updatedAt,
-            }
-
-            const unfixedFields = validateProtectedFields(
-                previousFormData,
-                unlockedFormData
-            )
-
-            if (unfixedFields.length !== 0) {
-                const errMessage = `Transient server error: attempted to modify un-modifiable field(s): ${unfixedFields.join(
-                    ','
-                )}.  Please refresh the page to continue.`
+            // If updatedAt does not match concurrent editing occurred.
+            if (
+                contractWithHistory.draftRevision.updatedAt !==
+                unlockedFormData.updatedAt
+            ) {
+                const errMessage = `Transient server error: Concurrent editing occurred. Please refresh the page to continue.`
```

Comment:
@macrael @haworku Could use some input in the error message here. I asked chatgpt to come up with some options

Optimistic Concurrency Control (commonly used with versioning):
1 ."Update failed: Record has been modified by another user."
2 ."Concurrent update error: The data you are trying to modify has changed since you last retrieved it."
3 ."Update conflict: Your changes were not saved because someone else has already updated this record."
4. "Version mismatch: The record has been updated by another process. Please refresh and try again."

Pessimistic Concurrency Control (commonly used with locks):
5. "Update failed: Record is locked by another user."
6. "Concurrency error: This data is currently being edited by another user. Please try again later."
7. "Locking conflict: Another user currently holds a lock on this record."
8. "Resource locked: Please wait for the other transaction to complete and try again."

General Concurrency Issues:
9. "Database contention: Too many concurrent updates. Please try again in a few moments."
10. "Concurrency violation: Simultaneous updates detected."
11. "Transaction collision: Another process has modified the data you are trying to update."
12. "Concurrent update error: Your operation could not be completed due to simultaneous updates by other users."

I like 2 with the addition of `Please refresh the page to continue.`

Reply:
We don't. We show the generic error banner.

Reply:
Ok then this is fine being technical